File: task-03/prime_number.java
import java.util.Scanner;

public class PrimeNumbers {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter a number (n): ");
        
        int n;
        while (true) {
            if (scanner.hasNextInt()) {
                n = scanner.nextInt();
                break;
            } else {
                System.out.println("Invalid input. Please enter a valid number.");
                scanner.nextLine();
            }
        }

        System.out.println("Prime numbers up to " + n + ":");
        for (int i = 2; i <= n; i++) {
            boolean isPrime = true;
            for (int j = 2; j < i; j++) {
                if (i % j == 0) {
                    isPrime = false;
                    break;
                }
            }
            if (isPrime) {
                System.out.print(i + " ");
            }
        }
        System.out.println();
        
        scanner.close();
    }
}

File: task-03/rust/prime_number.rs
use std::io;

fn main() {
    let mut input = String::new();

    println!("Enter a number (n):");

    io::stdin().read_line(&mut input).expect("Failed to read line");

    let n: u32 = input.trim().parse().expect("Invalid input. Please enter a valid number.");
    
    if n<2 {
        println!("There are no prime numbers below 2")
    } else {
    println!("Prime numbers up to {}:", n);
    }
    for num in 2..=n {
        let mut is_prime = true;
        for i in 2..num {
            if num % i == 0 {
                is_prime = false;
                break;
            }
        }
        if is_prime {
            print!("{} ", num);
        }
    }

    println!();
}

